reconcile/utils/instrumented_wrappers.py

```diff
@@ -0,0 +1,109 @@
+import os
+
+from prometheus_client import Counter, Gauge
+
+from sretoolbox.container import Image
+from sretoolbox.container import Skopeo
+
+# TODO: move these to a shared, constants module
+
+INTEGRATION_NAME = os.environ.get('INTEGRATION_NAME', '')
+SHARDS = os.environ.get('SHARDS', 1)
+SHARD_ID = int(os.environ.get('SHARD_ID', 0))
+
+
+class InstrumentedImage(Image):
+    """Normal Image that exposes the count of reachouts to external
+    registries.
+
+    It helps us understand the performance of our caches and predict
+    our mirroring-related costs.
+
+    """
+    _registry_reachouts = Counter(
+        name='qontract_reconcile_registry_reachouts',
+        documentation='Number of GET requests on public image registries',
+        labelnames=['integration', 'shard', 'shard_id'])
+
+    def _get_manifest(self):
+        self._registry_reachouts.labels(
+            integration=INTEGRATION_NAME,
+            shard=SHARDS,
+            shard_id=SHARD_ID
+        ).inc()
+        super()._get_manifest()
+
+
+class InstrumentedCache:
+    _cache_hits = Counter(
+        name='qontract_reconcile_cache_hits',
```

Comment:
Prometheus convention is for counters to end in `_total`
```suggestion
        name='qontract_reconcile_cache_hits_total',
```